analyses/malawi/dryspells_trigger/07_ecmwf_monthly_skill_general.md

```diff
@@ -41,164 +37,372 @@ reload(processing)
+# make new ax object for the cbar
+cbar_ax = g.fig.add_axes([.85, .25, .05, .4])  # x, y, width, height
+plt.colorbar(cax=cbar_ax)
+g.ax_joint.legend()
+g.fig.suptitle("Bias plot of observed vs forecasted values")
+g.fig.subplots_adjust(top=0.95) # Reduce plot to make room 
+```
+
+Since our main months of interest are January and February, we zoom in on these months. 
+From here we can see that the bias is a lot higher for these months compared to all months. Again for values up to 300 the forecast has a tendency to overpredict and for higher values to underpredict. 
+
+It is hard to say whether this increased bias is due to the period or the range of precipitation, while these are heavily intertwined. 
+
+```python
+#plot the observed vs forecast-observed to get a feeling for the discrepancy between the two
+df_forobs_selm=df_forobs[(df_forobs.time.dt.month.isin([1,2]))]
+g=sns.jointplot(data=df_forobs_selm,y="diff_forobs",x="precip_obs", kind="hex",height=10,joint_kws={ 'bins':'log'})
+#compute the average value of the difference between the forecasted and observed values
+#do this in bins cause else very noisy mean
+bins = np.arange(0,df_forobs_selm.precip_obs.max()+20,10)
+group = df_forobs_selm.groupby(pd.cut(df_forobs_selm.precip_obs, bins))
+plot_centers = (bins [:-1] + bins [1:])/2
+plot_values = group.diff_forobs.median()
+g.ax_joint.plot(plot_centers,plot_values,color="#C25048",label="median")
+g.set_axis_labels("Observed monthly precipitation (mm)", "Forecasted - Observed monthly precipitation (mm)", fontsize=12)
+plt.subplots_adjust(left=0.2, right=0.8, top=0.8, bottom=0.2)  # shrink fig so cbar is visible
+plt.subplots_adjust(left=0.2, right=0.8, top=0.8, bottom=0.2)  # shrink fig so cbar is visible
+# make new ax object for the cbar
+cbar_ax = g.fig.add_axes([.85, .25, .05, .4])  # x, y, width, height
+plt.colorbar(cax=cbar_ax)
+g.ax_joint.legend()
+g.fig.suptitle("Bias plot of observed vs forecasted values for January and February")
+g.fig.subplots_adjust(top=0.95)
+```
+
+Above we looked at the bias across all leadtimes and per cell. We now take the median of the error across all cells and months, but separated by leadtime. We can see that across all values the error is not that large, which was expected from the plots above
+
+
+Questions: 
+- better use mean or median to compute the difference between observation and forecast? As they give quite different results 
+- Does using the CI make sense or better use percentiles? Due to the large number of values the CI becomes small (which can be a good thing), but at the same time the std is quite large
+
+```python
+def calc_diff_stats(observations,forecast):
+    diff=forecast-observations
+    diff_mean=diff.mean()
+    diff_median=diff.median()
+    diff_std=diff.std()
+    diff_count=diff.count()
+    ci95_hi = diff_mean + 1.95*diff_std/math.sqrt(diff_count)
+    ci95_lo = diff_mean - 1.95*diff_std/math.sqrt(diff_count)
```

Comment:
The confidence interval seems a bit too small, I think because the points in your sample are not independent. The best way to get a confidence interval here would be to use the ensemble distribution

Reply:
Smart on the not independence! 
Regarding ensemble distribution, I am not sure. As we want to aggregate them to one number and here we want to know how well that aggregation works. Why would you say to use the ensemble distribution?

Reply:
Hm, as I'm still a bit confused about what's plotted here maybe we could talk about it